arraycontext/context.py

```diff
@@ -520,6 +521,8 @@ def compile(self, f: Callable[..., Any]) -> Callable[..., Any]:
         it may be called only once (or a few times).
 
         :arg f: the function executing the computation.
+        :arg single_version_only: If *True*, raise an error if *f* is compiled
+            more than once (due to different input argument types).
```

Comment:
- This needs to explain quite a bit more context, such as by explaining what triggers a recompile.
- I'm not sure I like "version" as a noun here.